### PR TITLE
add steps using LazyList

### PR DESCRIPTION
**Pull request by wtmount**

### CODE REVIEW

File: src/main/scala/collections/Steps.scala [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package collections
```
Comment (volodvro) [inline src/main/scala/collections/Steps.scala:1]: the code should be properly tested. Add unit tests, please

File: src/main/scala/collections/Steps.scala [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+object Steps extends App {
+  val N = 10
+  println(countSteps(N).size)
```
Comment (volodvro) [inline src/main/scala/collections/Steps.scala]: here you check the size of list and get 14 for 10-length list. Isn't it odd? Not mentioning the result sequence is full of 1s

File: src/main/scala/collections/Steps.scala [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+  def countSteps(n: Int): LazyList[Int] = {
```
Comment (volodvro) [inline src/main/scala/collections/Steps.scala]: you defined a function that produces a separate lazy list for each call, which is not efficient, rework it please to have a single lazy list for all calls

**Review by volodvro — CHANGES_REQUESTED** [review]

File: src/main/scala/collections/Steps.scala (hunk 1 above)
Comment (wtmount) [reply src/main/scala/collections/Steps.scala:1]: added unit tests

File: src/main/scala/collections/Steps.scala (hunk 3 above)
Comment (wtmount) [reply src/main/scala/collections/Steps.scala]: changed the function implementation

File: src/main/scala/collections/Steps.scala [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package collections
+
+class Steps {
+  val steps = List(2, 3, 5)
+
+  def countSteps(): LazyList[Int] = 1 #:: count(LazyList(1))
+
+  private def count(l: LazyList[Int] = LazyList.empty): LazyList[Int] = {
```
Comment (volodvro) [inline src/main/scala/collections/Steps.scala:8]: much better. The only thing, there is no need to make a Steps class, could be an object

Comment (wtmount) [reply src/main/scala/collections/Steps.scala:8]: made Steps an object

**Pull request merged**